Manifests itself as an issue only with clang+glibc
(gcc+glibc, or clang+musl are fine).

https://github.com/qt/qtlocation/commit/3a1b28126a0a182bef9fe0ccb85409fceb39d1f3
From: Tim Blechmann <tim@klingt.org>
Date: Thu, 20 Jul 2023 17:04:39 +0800
Subject: [PATCH] location: include missing header

std::unique_ptr requires <memory>

Pick-to: 6.5 6.6
--- a/src/location/maps/qgeocameratiles_p.h
+++ b/src/location/maps/qgeocameratiles_p.h
@@ -17,6 +17,8 @@
 #include <QtLocation/private/qlocationglobal_p.h>
 #include <QtCore/QScopedPointer>
 
+#include <memory>
+
 QT_BEGIN_NAMESPACE
 
 class QGeoCameraData;
